app/screens/LoginScreen/loginScreen.js

```diff
@@ -9,132 +9,103 @@ import {
 import * as EmailValidator from "email-validator";
-import styles from "./style";
 import { SocialIcon } from "react-native-elements";
+import styles from "./style";
+
 export default class LoginScreen extends Component {
   constructor(props) {
     super(props);
     this.state = {
       email: "",
-      Password: "",
+      password: "",
     };
   }
 
   componentDidMount() {
     var that = this;
-
-    auth.onAuthStateChanged(function (user) {
+    onAuthStateChanged(auth, function (user) {
       if (user) {
         that.redirectUser();
       }
     });
   }
 
-  login() {
-    let email = this.state.email;
-    let password = this.state.Password;
-
-    let { navigate } = this.props.navigation;
-
-    auth
-      .signInWithEmailAndPassword(email, password)
-      .then(function (data) {
-        navigate("App");
-      })
-      .catch(function (error) {
-        var errorMessage = error.message;
-        alert(errorMessage.toString());
-      });
-  }
-
   redirectUser() {
     const { navigate } = this.props.navigation;
     navigate("App");
   }
 
-  _signInAsync = async () => {
-    if (EmailValidator.validate(this.state.email) === true) {
-      if (this.state.Pasword != "") {
-        this.login();
-      } else {
-        alert("Enter the password");
-      }
+  // Validate the user's email and password
+  async _signInAsync() {
+    if (!EmailValidator.validate(this.state.email)) {
+      alert("Please enter a valid email!");
+    } else if (this.state.password == "") {
+      alert("Enter your password!");
     } else {
-      alert("Please enter A Valid Email");
+      this.login();
     }
-  };
-
-  onPressLogin() {
-    LoginManager.logInWithReadPermissions(["public_profile", "email"]).then(
-      result => this._handleCallBack(result),
-      function (error) {
-        alert("Login fail with error: " + error);
-      }
-    );
   }
 
-  _handleCallBack(result) {
-    let _this = this;
-    if (result.isCancelled) {
-      alert("Login cancelled");
-    } else {
-      AccessToken.getCurrentAccessToken().then(data => {
-        const token = data.accessToken;
-        fetch(
-          "https://graph.facebook.com/v2.8/me?fields=id,first_name,last_name,gender,birthday&access_token=" +
-            token
-        )
-          .then(response => response.json())
-          .then(json => {
-            const imageSize = 120;
-            const facebookID = json.id;
-            const fbImage = `https://graph.facebook.com/${facebookID}/picture?height=${imageSize}`;
-            this.authenticate(data.accessToken).then(function (result) {
-              const { uid } = result;
-              _this.createUser(uid, json, token, fbImage);
-            });
-          })
-          .catch(function (err) {
-            console.log(err);
-          });
-      });
+  // Signin with email and password
+  async login() {
+    try {
+      let email = this.state.email;
+      let password = this.state.password;
```

Comment:
Why not "const"?

Reply:
There is no specfic reason of using "let" both const and let work similar in a block scope. As let was already present/used there so left it as it is.

Reply:
Converted let to const